src/control/lib/control/system.go

```diff
@@ -201,3 +203,75 @@ func ListPools(ctx context.Context, rpcClient UnaryInvoker, req *ListPoolsReq) (
 	resp := new(ListPoolsResp)
 	return resp, convertMSResponse(ur, resp)
 }
+
+// SystemRanksReq contains the parameters for a system ranks request.
+type SystemRanksReq struct {
+	unaryRequest
+	Ranks []system.Rank
+}
+
+// SystemRanksResp contains the response from a system ranks request.
+type SystemRanksResp struct {
+	HostErrorsResp // record unresponsive hosts
+	RankResults    system.MemberResults
+}
+
+// addHostResponse is responsible for validating the given HostResponse
+// and adding it's results to the SystemRanksResp.
+func (srr *SystemRanksResp) addHostResponse(hr *HostResponse) (err error) {
+	pbResp, ok := hr.Message.(*mgmtpb.RanksResp)
+	if !ok {
+		return errors.Errorf("unable to unpack message: %+v", hr.Message)
+	}
+
+	memberResults := make(system.MemberResults, 0)
+	if err := convert.Types(pbResp.GetResults(), &memberResults); err != nil {
+		if srr.HostErrors == nil {
+			srr.HostErrors = make(HostErrorsMap)
+		}
+		return srr.HostErrors.Add(hr.Addr, err)
+	}
+
+	if srr.RankResults == nil {
+		srr.RankResults = make(system.MemberResults, 0)
+	}
+	srr.RankResults = append(srr.RankResults, memberResults...)
```

Comment:
For the record, append with a nil slice works fine: https://play.golang.org/p/u3Bnmwrs_sp

This pattern makes sense in the other `addHostResponse()` helpers because we have to make the map before inserting anything.

Reply:
done